go/systests/ephemeral_test.go

```diff
@@ -48,3 +48,64 @@ func checkNewTeamEKNotifications(tc *libkb.TestContext, notifications *teamNotif
 		}
 	}
 }
+
+func TestTeamTransactionWithTeamEK(t *testing.T) {
+	runTeamTransaction(t, true /* createTeamEK*/)
+}
+
+func TestTeamTransactionNoTeamEK(t *testing.T) {
+	runTeamTransaction(t, false /* createTeamEK*/)
+}
+
+func getTeamEK(g *libkb.GlobalContext, teamID keybase1.TeamID, generation keybase1.EkGeneration) (keybase1.TeamEk, error) {
+	storage := g.GetTeamEKBoxStorage()
+	return storage.Get(context.Background(), teamID, generation)
+}
+
+func runTeamTransaction(t *testing.T, createTeamEK bool) {
+	ctx := newSMUContext(t)
+	defer ctx.cleanup()
+
+	ann := ctx.installKeybaseForUser("ann", 10)
+	ann.signup()
+	bob := ctx.installKeybaseForUser("bob", 10)
+	bob.signup()
+
+	annG := ann.getPrimaryGlobalContext()
+	ephemeral.ServiceInit(annG)
+	bobG := bob.getPrimaryGlobalContext()
+	ephemeral.ServiceInit(bobG)
+
+	team := ann.createTeam([]*smuUser{})
+	teamName, err := keybase1.TeamNameFromString(team.name)
+	require.NoError(t, err)
+	teamID := teamName.ToPrivateTeamID()
+
+	var expectedMetadata keybase1.TeamEkMetadata
+	var expectedGeneration keybase1.EkGeneration
+	if createTeamEK {
+		ekLib := annG.GetEKLib()
+		teamEK, err := ekLib.GetOrCreateLatestTeamEK(context.Background(), teamID)
+		require.NoError(t, err)
+
+		expectedMetadata = teamEK.Metadata
+		expectedGeneration = expectedMetadata.Generation
+	} else {
+		expectedMetadata = keybase1.TeamEkMetadata{}
+		expectedGeneration = 1
+	}
+
+	ann.addTeamMember(team, bob, keybase1.TeamRole_WRITER)
+
+	annTeamEK, annErr := getTeamEK(annG, teamID, expectedGeneration)
+	bobTeamEK, bobErr := getTeamEK(bobG, teamID, expectedGeneration)
+	if createTeamEK {
+		require.NoError(t, annErr)
+		require.NoError(t, bobErr)
+	} else {
+		require.Error(t, annErr)
+		require.Error(t, bobErr)
```

Comment:
Can we be specific about the error type here?

Reply:
yeah we can use the `EKUnboxErr` since that's waht we expect

Reply:
oops actually we don't have a special error for this -- there is nothing on the server to retrieve.. we can revisit later if we want to be more specific